src/main/java/it/pagopa/pn/deliverypush/action/utils/AttachmentUtils.java

```diff
@@ -83,19 +82,20 @@ private void forEachAttachment(NotificationInt notification, Consumer<Notificati
 
     private void checkAttachment(NotificationDocumentInt attachment) {
         NotificationDocumentInt.Ref ref = attachment.getRef();
-
-        FileDownloadResponseInt fd = safeStorageService.getFile(ref.getKey(),true);
+        FileDownloadResponseInt fd = null;
+        try {
+            fd = safeStorageService.getFile(ref.getKey(),true);
+        } catch ( PnNotFoundException ex ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)